File: code/stable/bmbnd/src/java/BmBnd/InputParameters.java
package BmBnd;

/** \file InputParameters.java
    \author Jason Balaci
    \brief Provides the function for reading inputs
*/
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class InputParameters {
    
    /** \brief Reads input from a file with the given file name
        \param filename name of the input file
        \return array containing the following values:
        \return coefficient of w_B's term of power 0 (N/m)
        \return coefficient of w_B's term of power 1 (N/m^2)
        \return coefficient of w_B's term of power 2 (N/m^3)
        \return coefficient of w_B's term of power 3 (N/m^4)
        \return length of the beam (m)
        \return modulus of elasticity of the beam (Pa)
        \return moment of second area of a cross-section of the beam (m)
    */
    public static Object[] get_input(String filename) throws FileNotFoundException {
        double a_0;
        double a_1;
        double a_2;
        double a_3;
        double L_B;
        double E_B;
        double I_B;
        
        Scanner infile;
        infile = new Scanner(new File(filename));
        infile.nextLine();
        a_0 = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        a_1 = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        a_2 = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        a_3 = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        L_B = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        E_B = Double.parseDouble(infile.nextLine());
        infile.nextLine();
        I_B = Double.parseDouble(infile.nextLine());
        infile.close();
        
        Object[] outputs = new Object[7];
        outputs[0] = a_0;
        outputs[1] = a_1;
        outputs[2] = a_2;
        outputs[3] = a_3;
        outputs[4] = L_B;
        outputs[5] = E_B;
        outputs[6] = I_B;
        return outputs;
    }
}

File: code/stable/bmbnd/src/swift/InputParameters.swift
/** InputParameters.swift
    Provides the function for reading inputs
    - Authors: Jason Balaci
*/
import Foundation

extension String: Error {}

/** Reads input from a file with the given file name
    - Parameter filename: name of the input file
    - Returns: coefficient of w_B's term of power 0 (N/m)
    - Returns: coefficient of w_B's term of power 1 (N/m^2)
    - Returns: coefficient of w_B's term of power 2 (N/m^3)
    - Returns: coefficient of w_B's term of power 3 (N/m^4)
    - Returns: length of the beam (m)
    - Returns: modulus of elasticity of the beam (Pa)
    - Returns: moment of second area of a cross-section of the beam (m)
*/
func get_input(_ filename: String) throws -> (Double, Double, Double, Double, Double, Double, Double) {
    var a_0: Double
    var a_1: Double
    var a_2: Double
    var a_3: Double
    var L_B: Double
    var E_B: Double
    var I_B: Double
    
    var infile: URL
    infile = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.appendingPathComponent(filename)
    var goolContents: [[String]]
    do {
        goolContents = try String(contentsOf: infile).components(separatedBy: "\n").map({(l: String) -> [String] in l.components(separatedBy: " ")})
    } catch {
        throw "Error reading from file."
    }
    a_0 = Double(goolContents[1][0])!
    a_1 = Double(goolContents[2][0])!
    a_2 = Double(goolContents[3][0])!
    a_3 = Double(goolContents[4][0])!
    L_B = Double(goolContents[5][0])!
    E_B = Double(goolContents[6][0])!
    I_B = Double(goolContents[7][0])!
    
    return (a_0, a_1, a_2, a_3, L_B, E_B, I_B)
}
